lib/elastic_record/as_document.rb

```diff
@@ -1,7 +1,7 @@
 module ElasticRecord
   module AsDocument
     def as_search_document
-      doctype.mapping[:properties].each_with_object({}) do |(field, mapping), result|
+      doctype.mapping[:properties].each_with_object({}.with_indifferent_access) do |(field, mapping), result|
```

Comment:
There's gotta be a better way to deal with mixed hash key types instead of peppering `.with_indifferent_access` everywhere.

Reply:
Originally, I was using `deep_symbolize_keys` in the accessor methods for `mapping`. But using `with_indifferent_access` was per @matthuhiggins. I'm open to better ways though.

Reply:
I meant for `with_indifferent_access` on the `mapping` and `setting` instances, not on calls to the instance.